src/main/java/edu/rpi/legup/puzzle/treetent/rules/LinkTreeCaseRule.java

```diff
@@ -152,4 +153,42 @@ public String checkRuleRaw(TreeTransition transition) {
     public String checkRuleRawAt(TreeTransition transition, PuzzleElement puzzleElement) {
         return checkRuleRaw(transition);
     }
+
+    /**
+     * Returns the elements necessary for the cases returned by getCases(board,puzzleElement) to be valid
+     * Overridden by case rules dependent on more than just the modified data
+     *
+     * @param board         board state at application
+     * @param puzzleElement selected puzzleElement
+     * @return List of puzzle elements (typically cells) this application of the case rule depends upon.
+     * Defaults to any element modified by any case
+     */
+    @Override
+    public List<PuzzleElement> dependentElements(Board board, PuzzleElement puzzleElement){
+        List<PuzzleElement> elements = new ArrayList<>(List.of(board.getPuzzleElement(puzzleElement)));
+
+        TreeTentBoard treeTentBoard = (TreeTentBoard) board;
+        TreeTentCell point = (TreeTentCell) puzzleElement;
+
+        //get all adjacent cells
```

Comment:
While is is technically correct, getting each adjacent cell is using duplicate logic. Consider using a `for` loop instead.

Some pseudocode to help:
```java
List<Point> directions = { new Point(0, 1), ..., new Point(-1, 0) };
for (Point direction : directions) {
        TreeTentCell adj = treeTentBoard.getCell(loc.x + point.x, loc.y + point.y);
        ...
        }
```

Reply:
This was copied from TreeTendBoard.getAdjacent, changed both to a for loop.